R/type-vctr.R

```diff
@@ -56,20 +56,37 @@ new_vctr <- function(.data, ..., class = character()) {
   if (!is_vector(.data)) {
     abort("`.data` must be a vector type.")
   }
-  .data <- validate_attr(.data)
+  nms <- validate_names(.data)
 
-  structure(.data, ..., class = c(class, "vctrs_vctr"))
+  vec_structure(.data, names = nms, ..., class = c(class, "vctrs_vctr"))
 }
 
-validate_attr <- function(.data) {
+vec_structure <- function(.data, ...) {
```

Comment:
Should this be called `restructure()`? This way the name hints at the removal of existing attributes.